### ISSUE TITLE
`piperider init` command fails when macros are present in `dbt_project.yml`

### ISSUE THREAD

**Issue by waligob** [InfuseAI/piperider#665]
**Describe the bug**
The `piperider init` command doesn’t complete successfully when there’s jinja in the `dbt_project.yml` file (even though it looks like this was fixed in #658. Instead it yields an error like:
```
Error: Failed to load dbt project '<project_yml_filepath>'. 
 Reason: <macro_name> is undefined 
``` 

**To Reproduce**
1. Setup new virtual environment and run `pip install "piperider[bigquery]==0.23.0"`
2. Navigate to the root directory of a dbt project whose `dbt_project.yml` file contains jinja
3. Execute `piperider init` and select the appropriate `dbt_project.yml` filepath

**Expected behavior**
Expected output 
```
[ DBT ] Use the existing dbt project file: 
<project_yml_filepath>
[ DBT ] By default, PipeRider will profile the models and metrics
with 'piperider' tag
        Apply 'piperider' tag to your models or change the tag in
'.piperider/config.yml'

───────────────────── .piperider/config.yml ─────────────────────
   1 dataSources: []                                             
   2 dbt:                                                        
   3   projectDir: .                                             
   4   tag: piperider                                            
   5                                                             
   6 profiler:                                                   
   7 #   table:                                                  
   8 #     # the maximum row count to profile. (Default unlimited
   9 #     limit: 1000000                                        
  10 #     duplicateRows: false                                  
  11                                                             
  12 telemetry:                                                  
  13   id: <telemetry_id>                      
  14                                                             
───────────────── End of .piperider/config.yml ──────────────────
──────────── Recipe: .piperider/compare/default.yml ─────────────
   1 base:                                                       
   2   branch: main                                              
   3   dbt:                                                      
   4     commands:                                               
   5     - dbt deps                                              
   6     - dbt build                                             
   7   piperider:                                                
   8     command: piperider run                                  
   9 target:                                                     
  10   dbt:                                                      
  11     commands:                                               
  12     - dbt deps                                              
  13     - dbt build                                             
  14   piperider:                                                
  15     command: piperider run                                  
  16                                                             
───────────────────────── End of Recipe ─────────────────────────

Next step:
  Please execute command 'piperider diagnose' to verify 
configuration
```

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Debian GNU/Linux
 - Python 3.9.16
 - dbt 1.4.5 

**Additional context**
In our particular instance, `dbt_project.yml` contains three macros:
* A custom macro used to set a variable value:
```
vars:
  run_date: "{{ get_current_date() }}"
```
* Two nested built-in macros to configure an on-run-end hook:
```
on-run-end:
  - "{% if target.name == 'prod' %}{{ dbt_artifacts.upload_results(results) }}{% endif %}"
```
If any of these macros are present, the `pipeline init` command fails.

**Comment by DaveFlynn** [comment]
Thanks for submitting the issue @waligob this will be addressed in a point release very soon. I will update once the fix is released.

**Comment by waligob** [comment]
This appears to be fixed in `0.23.1`  (via #664)

**Comment by clkao** [comment]
closed by #670
